ui/infra-compare/InfraCompare.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import withValidation from '../perfherder/Validation';
+
+import { getCounterMap } from './helpers';
+import { phTimeRanges } from './constants';
+import InfraCompareTableView from './InfraCompareTableView';
+
+class InfraCompareView extends React.PureComponent {
+  getInterval = (oldTimestamp, newTimestamp) => {
+    const now = new Date().getTime() / 1000;
+    let timeRange = Math.min(oldTimestamp, newTimestamp);
+    timeRange = Math.round(now - timeRange);
+    const newTimeRange = phTimeRanges.find((time) => timeRange <= time.value);
+    return newTimeRange.value;
+  };
+
+  queryParams = (project, interval) => ({
+    project,
+    interval,
+  });
+
+  getQueryParams = (timeRange) => {
+    const {
+      originalProject,
+      newProject,
+      originalRevision,
+      newRevision,
+      newResultSet,
+      originalResultSet,
+    } = this.props.validated;
+
```

Comment:
We have a styling idiom that we try to follow, though it is not always followed.  We group declarations together, then have a blank line between declarations and functional code.  So here, we would have no blank line between the `const` and the `let`s.  But we DO have a blank line between the `let` lines and the `if` line below.  Not a huge deal, but as much as you can, it would be nice to follow this convention.  :)